src/dashboards/components/dashboard_index/DashboardsIndexContentsPaginated.tsx

```diff
@@ -0,0 +1,208 @@
+  public currentPage: number = 1
+  public rowsPerPage: number = 12
+  public totalPages: number
+
+  private memGetSortedResources = memoizeOne<typeof getSortedResources>(
+    getSortedResources
+  )
+  constructor(props) {
+    super(props)
+    this.paginationRef = createRef<HTMLDivElement>()
+  }
+
+  public componentDidMount() {
+    const {dashboards} = this.props
+
+    const dashboardIDs = dashboards.map(d => d.id)
+    this.props.retainRangesDashTimeV1(dashboardIDs)
+    this.props.checkDashboardLimits()
+
+    const params = new URLSearchParams(window.location.search)
+    const urlPageNumber = parseInt(params.get('page'), 10)
+
+    const passedInPageIsValid =
+      urlPageNumber && urlPageNumber <= this.totalPages && urlPageNumber > 0
+    if (passedInPageIsValid) {
+      this.currentPage = urlPageNumber
+    }
+  }
+
+  public componentDidUpdate() {
+    // if the user filters the list while on a page that is
+    // outside the new filtered list put them on the last page of the new list
+    if (this.currentPage > this.totalPages) {
+      this.paginate(this.totalPages)
+    }
+  }
+
+  public paginate = page => {
+    this.currentPage = page
+    const url = new URL(location.href)
+    url.searchParams.set('page', page)
+    history.replaceState(null, '', url.toString())
+    this.forceUpdate()
+  }
+
+  public renderDashboardCards() {
+    const sortedDashboards = this.memGetSortedResources(
+      this.props.dashboards,
+      this.props.sortKey,
+      this.props.sortDirection,
+      this.props.sortType
+    )
+
+    const startIndex = this.rowsPerPage * Math.max(this.currentPage - 1, 0)
+    const endIndex = Math.min(
+      startIndex + this.rowsPerPage,
+      this.props.totalDashboards
+    )
+    const rows = []
+    for (let i = startIndex; i < endIndex; i++) {
+      const dashboard = sortedDashboards[i]
+      if (dashboard) {
+        rows.push(dashboard)
+      }
+    }
+    return rows
+  }
+
+  public render() {
+    const {
+      searchTerm,
+      status,
+      dashboards,
+      onFilterChange,
+      onCreateDashboard,
+    } = this.props
+
+    const heightWithPagination =
+      this.paginationRef?.current?.clientHeight ||
+      DEFAULT_PAGINATION_CONTROL_HEIGHT
+    const height = this.props.pageHeight - heightWithPagination
+
+    this.totalPages = Math.max(
+      Math.ceil(dashboards.length / this.rowsPerPage),
+      1
+    )
+
+    if (status === RemoteDataState.Done && !dashboards.length) {
+      return (
+        <DashboardsTableEmpty
+          searchTerm={searchTerm}
+          onCreateDashboard={onCreateDashboard}
+          summonImportFromTemplateOverlay={this.summonImportFromTemplateOverlay}
+          summonImportOverlay={this.summonImportOverlay}
+        />
+      )
+    }
+
+    return (
+      <>
+        <ResourceList style={{width: this.props.pageWidth}}>
+          <ResourceList.Body
+            style={{maxHeight: height, minHeight: height, overflow: 'auto'}}
```

Comment:
cool, this `overflow: 'auto'` avoids the [bug](https://github.com/influxdata/ui/issues/3324) I keep seeming to stumble over.